pkg/collector/harbor_events.go

```diff
@@ -0,0 +1,508 @@
+		if scanOverview["scan_status"].(string) == "Success" {
+			status = discovery.Discovered_FINISHED_SUCCESS
+			vulnerabilityDetails = t.getVulnerabilityDetails(scanOverview["severity"].(string))
+		} else if scanOverview["scan_status"].(string) == "Error" {
+			status = discovery.Discovered_FINISHED_FAILED
+		}
+
+		discoveryDetails := &grafeas.Occurrence_Discovered{
+			Discovered: &discovery.Details{
+				Discovered: &discovery.Discovered{
+					AnalysisStatus: status,
+				},
+			},
+		}
+
+		o := newHarborImageScanOccurrence(resources[i], t.project)
+		o.Details = discoveryDetails
+		occurrences = append(occurrences, o)
+		if scanOverview["scan_status"].(string) == "Success" {
+			o = newHarborImageScanOccurrence(resources[i], t.project)
+			o.Details = vulnerabilityDetails
+			occurrences = append(occurrences, o)
+		}
+
+	}
+	return occurrences
+}
+
+func newHarborImageScanOccurrence(resource *Resource, projectName string) *grafeas.Occurrence {
+	o := &grafeas.Occurrence{
+		Resource: &grafeas.Resource{
+			Uri: HarborOccurrenceResourceURI(resource.ResourceURL, resource.Digest),
+		},
+		NoteName: HarborOccurrenceNote(projectName),
+	}
+	return o
+}
+
+func HarborOccurrenceResourceURI(url, digest string) string {
+	return fmt.Sprintf("%s@%s", url, digest)
+}
+
+func HarborOccurrenceNote(projectName string) string {
+	return fmt.Sprintf("projects/%s/notes/%s", "rode", projectName)
+}
+
+func (t *HarborEventCollector) getVulnerabilityDetails(severity string) *grafeas.Occurrence_Vulnerability {
+	vulnerabilitySeverity := t.getVulnerabilitySeverity(severity)
+	vulnerabilityDetails := &grafeas.Occurrence_Vulnerability{
+		Vulnerability: &vulnerability.Details{
+			Severity: vulnerabilitySeverity,
+			PackageIssue: []*vulnerability.PackageIssue{
+				{
+					AffectedLocation: &vulnerability.VulnerabilityLocation{
+						CpeUri:  "TODO",
+						Package: "TODO",
+						Version: &packag.Version{
+							Kind: packag.Version_NORMAL,
+							Name: "TODO",
+						},
+					},
+				},
+			},
+		},
+	}
+	return vulnerabilityDetails
+}
+
+func (t *HarborEventCollector) getVulnerabilitySeverity(v string) vulnerability.Severity {
+	switch v {
+	case HarborSeverityCritical:
+		return vulnerability.Severity_CRITICAL
+	case HarborSeverityHigh:
+		return vulnerability.Severity_HIGH
+	case HarborSeverityMedium:
+		return vulnerability.Severity_MEDIUM
+	case HarborSeverityLow:
+		return vulnerability.Severity_LOW
+	case HarborSeverityNegligible:
+		return vulnerability.Severity_MINIMAL
+	case HarborSeverityNone:
+		return vulnerability.Severity_MINIMAL
+	default:
+		return vulnerability.Severity_SEVERITY_UNSPECIFIED
+	}
+}
+
+// Assumes Harbor admin creds are deployed in the same namespace as the Collector CR
+func (t *HarborEventCollector) getHarborCredentials(ctx context.Context, secretname string, namespace string) (string, error) {
+	config, err := rest.InClusterConfig()
```

Comment:
The manager comes with its own kubernetes client, and I think we should be using that instead of building our own using client-go.  See here as an example: https://github.com/liatrio/rode/blob/72f68cd4ce8c0bdb1b275328f3592a4cd7c07283/controllers/collector_controller.go#L70

We can pass this client as a reference to the harbor collector so you have access to it here, perhaps via `t.client`